addons/html_editor/static/src/others/dynamic_placeholder_plugin.js

```diff
@@ -73,14 +73,33 @@ export class DynamicPlaceholderPlugin extends Plugin {
     /**
      * @param {string} chain
      * @param {string} defaultValue
+     * @param {string} fieldType
      */
-    onValidate(chain, defaultValue) {
+    async onValidate(chain, defaultValue, fieldType) {
         if (!chain) {
             return;
         }
 
+        const fields = await this.services.field.loadFields(this.defaultResModel);
+        const partnerFields = Object.values(fields).filter(field => field.type === 'many2one' && field.relation === 'res.partner');
+
+        // Prioritize 'partner_id' if present, otherwise we'll use the first available field
+        const selectedField = partnerFields.find(field => field.name === 'partner_id') || partnerFields[0];
+
+        let dynamicPlaceholder;
+        if (fieldType == 'datetime') {
+            dynamicPlaceholder = selectedField ? `format_datetime(object.${chain}, tz=object.${selectedField.name}.tz)`
```

Comment:
Also note that we have ``_mail_get_timezone`` available, which selects timezone based on business record (e.g. event that have a date_tz field). Issue is that it is not easy to know which one to use ... maybe we should just always use partner one, then people can change if necessary.